src/include/duckdb/main/relation/table_relation.hpp

```diff
@@ -15,7 +15,8 @@ namespace duckdb {
 
 class TableRelation : public Relation {
 public:
-	TableRelation(const shared_ptr<ClientContext> &context, unique_ptr<TableDescription> description);
+	TableRelation(const shared_ptr<ClientContext> &context, unique_ptr<TableDescription> description,
```

Comment:
Rather than passing `acquire_lock` here, can we have a second constructor that allows passing in the `ClientContextWrapper` directly? We can then overload the `RelationContextWrapper` with an `InternalRelationContextWrapper` which does not lock in `TryBindRelation`